tools/make_patch.c

```diff
@@ -0,0 +1,437 @@
+		buffer_append(patches, &(struct Patch){current_offset, length});
+		bool modified = false;
+		if (length == 1) {
+			int c = getc(new_rom);
+			modified = c != getc(orig_rom);
+			fprintf(output, isupper((unsigned char)command[0]) ? "0x%02X" : "0x%02x", c);
+		} else {
+			fprintf(output, command[strlen(command) - 1] == '_' ? "a%d: " : "a%d:", length);
+			for (int i = 0; i < length; i++) {
+				if (i) {
+					putc(' ', output);
+				}
+				int c = getc(new_rom);
+				modified |= c != getc(orig_rom);
+				fprintf(output, isupper((unsigned char)command[0]) ? "%02X" : "%02x", c);
+			}
+		}
+		if (!modified) {
+			fprintf(stderr, PROGRAM_NAME ": Warning: \"vc_patch %s\" doesn't alter the ROM\n", current_hook->name);
+		}
+
+	} else if (!strcmp(command, "dws") || !strcmp(command, "DWS") || !strcmp(command, "dws_") || !strcmp(command, "DWS_")) {
+		if (argc < 1) {
+			error_exit("Error: Invalid arguments for command: \"%s\"", command);
+		}
+		fprintf(output, command[strlen(command) - 1] == '_' ? "a%d: " : "a%d:", argc * 2);
+		for (int i = 0; i < argc; i++) {
+			int value = parse_arg_value(argv[i], false, symbols, current_hook->name);
+			if (value > 0xffff) {
+				error_exit("Error: Invalid value for \"%s\" argument: 0x%x", command, value);
+			}
+			if (i) {
+				putc(' ', output);
+			}
+			fprintf(output, isupper((unsigned char)command[0]) ? "%02X %02X": "%02x %02x", value & 0xff, value >> 8);
+		}
+
+	} else if (!strcmp(command, "db") || !strcmp(command, "DB") || !strcmp(command, "db_") || !strcmp(command, "DB_")) {
+		if (argc != 1) {
+			error_exit("Error: Invalid arguments for command: \"%s\"", command);
+		}
+		int value = parse_arg_value(argv[0], false, symbols, current_hook->name);
+		if (value > 0xff) {
+			error_exit("Error: Invalid value for \"%s\" argument: 0x%x", command, value);
+		}
+		fputs(command[strlen(command) - 1] == '_' ? "a1: " : "a1:", output);
+		fprintf(output, isupper((unsigned char)command[0]) ? "%02X" : "%02x", value);
+
+	} else if (!strcmp(command, "hex") || !strcmp(command, "HEX") || !strcmp(command, "HEx") || !strcmp(command, "Hex") || !strcmp(command, "heX") || !strcmp(command, "hEX")) {
+		if (argc != 1 && argc != 2) {
+			error_exit("Error: Invalid arguments for command: \"%s\"", command);
+		}
+		int value = parse_arg_value(argv[0], true, symbols, current_hook->name);
+		int padding = argc > 1 ? parse_number(argv[1], 0) : 2;
+		if (!strcmp(command, "HEx")) {
+			fprintf(output, "0x%0*X%02x", padding - 2, value >> 8, value & 0xff);
+		} else if (!strcmp(command, "Hex")) {
+			fprintf(output, "0x%0*X%03x", padding - 3, value >> 12, value & 0xfff);
+		} else if (!strcmp(command, "heX")) {
+			fprintf(output, "0x%0*x%02X", padding - 2, value >> 8, value & 0xff);
+		} else if (!strcmp(command, "hEX")) {
+			fprintf(output, "0x%0*x%03X", padding - 3, value >> 12, value & 0xfff);
+		} else {
+			fprintf(output, isupper((unsigned char)command[0]) ? "0x%0*X" : "0x%0*x", padding, value);
+		}
+
+	} else {
```

Comment:
Where's the comment (`{{#: This is a comment}}`) command?

Reply:
Do any lines *in* the template file need comments?

Reply:
Nope; we removed them for a reason. In order to implement them.. it would get confusing between custom comments and original comments.

Reply:
> Do any lines in the template file need comments?

I'd say yes, or at least it's a good tool to have to help document the patches themselves.
We removed the only comment in the original file since it explained the patch *format*, that has moved into a separate file and makes more sense there.